File: docs/vsto/codesnippet/CSharp/trin_excelworkbookpda/registrypda/class1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
//<snippet4>
using Microsoft.VisualStudio.Tools.Applications.Deployment;
//</snippet4>

namespace RegistryPDA 
{
//<snippet5>
    public class RegistryPDA : IAddInPostDeploymentAction
// </snippet5>
    {
//<snippet6>
        public void Execute(AddInPostDeploymentActionArgs args)
        {

            switch (args.InstallationStatus)
            {
                case AddInInstallationStatus.InitialInstall:
                    Microsoft.Win32.RegistryKey key;
                    key = Microsoft.Win32.Registry.LocalMachine.CreateSubKey("SOFTWARE\\Microsoft\\VSTO Runtime Setup\\v4");
                    key.SetValue("RegKeyString", "Post-Deployment Action Test");
                    key.Close();
                    break;
                case AddInInstallationStatus.Uninstall:
                    Microsoft.Win32.Registry.LocalMachine.DeleteSubKey("SOFTWARE\\Microsoft\\VSTO Runtime Setup\\v4\\RegKeyString");
                    break;
            }
        }
//</snippet6>
    }
}

File: docs/vsto/codesnippet/VisualBasic/trin_excelworkbookpda/registrypda/class1.vb
'<snippet4>
Imports Microsoft.VisualStudio.Tools.Applications.Deployment
'</snippet4>

'<snippet5>
Public Class RegistryPDA
    Implements IAddInPostDeploymentAction
'</snippet5>

'<snippet6>
    Sub Execute(ByVal args As AddInPostDeploymentActionArgs) Implements IAddInPostDeploymentAction.Execute
        Select Case args.InstallationStatus
            Case AddInInstallationStatus.InitialInstall
                Dim key As Microsoft.Win32.RegistryKey
                key = Microsoft.Win32.Registry.LocalMachine.CreateSubKey("SOFTWARE\Microsoft\VSTO Runtime Setup\v4")
                key.SetValue("RegKeyString", "Post-Deployment Action Test")
                key.Close()
                Exit Select
            Case AddInInstallationStatus.Uninstall
                Microsoft.Win32.Registry.LocalMachine.DeleteSubKey("SOFTWARE\Microsoft\VSTO Runtime Setup\v4\RegKeyString")
                Exit Select
        End Select
    End Sub
    '</snippet6>

End Class
